server/pulp/server/db/model/__init__.py

```diff
@@ -647,6 +648,36 @@ class CeleryBeatLock(Document):
     _ns = StringField(default='celery_beat_lock')
 
 
+class Digest(EmbeddedDocument):
+    """
+    Embedded digest (checksum) document.
+
+    :ivar algorithm: The digest algorithm.
+    :type algorighm: str
+    :ivar value: The digest value.
+    :type value: str
+    """
+    ALG_REGEX = r'({a})'.format(a='|'.join(hashlib.algorithms))
```

Comment:
Sadly, hashlib.algorithms is new in 2.7. I'm not sure if there's a good replacement that is 2.6 compatible.